prometheus/collector_metric.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Percona LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import "time"
+
+var timeToCollectDesc = NewDesc(
```

Comment:
I think it can be included in the MeasureCollectTime function and convert variable labels into constant labels.

Reply:
in fact it was crucial to move `collector` label to constLabel, because it will be used to calculate ID.
and as a result, if 2 collectors report different const labels they will have same ID

and it these to will be in cache it will fail with something like:
```
--- FAIL: TestMongoS (167.46s)
panic: descriptor Desc{fqName: "collector_scrape_time_ms", help: "Time taken for scrape by collector", constLabels: {}, variableLabels: [exporter collector]} already exists with the same fully-qualified name and const label values [recovered]
	panic: descriptor Desc{fqName: "collector_scrape_time_ms", help: "Time taken for scrape by collector", constLabels: {}, variableLabels: [exporter collector]} already exists with the same fully-qualified name and const label values
```

Reply:
TBH, I didn't understand what you wrote :) Your test was failing because both labels were variable labels. I would recommend making the `exporter` label constant as you did for the `collector` label.

Reply:
we can talk tomorrow.
If I change as you propose it will fail
![image](https://user-images.githubusercontent.com/91831719/176796755-29a81a65-59aa-4fb0-b4d0-a2569adf8f5a.png)

as it implemented now: collector - is a variable (as we have different ones); but exporter is a constant (in this case mongodb)

but! for some reason it is variable part is called "constant"... to be honest I have no idea why.
https://github.com/Percona-Lab/client_golang/blob/ea5a3aa8c041acc9632ce1a1994ecd8473c89546/prometheus/registry.go#L297

Reply:
That's not what I suggest :) if you open https://github.com/Percona-Lab/client_golang/blob/0a025dba3905a4912592fa44a106562168a34a39/prometheus/desc.go#L82 you will see that the last argument is a constLabels, so I suggested to make both labels `constant`.
Everything is fine now, we can go with a current solution.